code/game/gamemodes/game_mode.dm

```diff
@@ -186,6 +186,14 @@
 		return TRUE
 	if(station_was_nuked)
 		return TRUE
+
+	if(config.check_station_integrity)
+		var/static/datum/station_state/end_state = new /datum/station_state() //Static so it doesn't have to keep being remade.
+		end_state.count()
+		var/station_integrity = min(PERCENT(GLOB.start_state.score(end_state)), 100)
+		if(station_integrity <= 20) //Below 20% integrity
+			message_admins("The station/ship has suffered critical damage, with the config set so that it ends the round with critical damage. Doing so..")

```

Comment:
This is a little awkwardly phrased. I don't think you need to expressly tell the admins that the round is going to end due to a config setting. I think it's obvious enough from the fact that they're being messaged about it. Also, it's always going to be a ship, so you might as well specify as such.

Maybe: "The ship has sustained critical damage! Ending the round."

Reply:
That does sound better